doc/developers/contributing.rst

```diff
@@ -953,10 +953,14 @@ All logic behind estimator parameters,
 like translating string arguments into functions, should be done in ``fit``.
 
 Parameters can be passed to the ``fit`` method. Generally, these parameters
-should be restricted to variables that have the shape ``n_samples``.
-This allows for all relevant parameters to be sliced cleanly during
-cross-validation. All other parameters should be passed to
-``__init__`` or ``set_params``.
+should be restricted to variables that need to be sliced during
+cross-validation. Consequently, these variables would be either arrays with shape=[N] or
+array-like with shape=[N,D] where N is the number of samples. An example of
+this type of variable would be
+``sample_weight``. This interface allows all data-dependent
+parameters to be sliced cleanly during cross-validation.
+All other parameters should be passed to
+``__init__`` or set after construction using ``set_params``.
```

Comment:
Ok, understood. Thank you for your suggestions Nelson!

On Aug 7, 2016 4:31 PM, "Nelson Liu" <notifications@github.com> wrote:

> In doc/developers/contributing.rst
> <https://github.com/scikit-learn/scikit-learn/pull/7156#discussion_r73808517>
> :
>
> > @@ -953,10 +953,14 @@ All logic behind estimator parameters,
> >  like translating string arguments into functions, should be done in ``fit``.
> >
> >  Parameters can be passed to the ``fit`` method. Generally, these parameters
> > -should be restricted to variables that have the shape ``n_samples``.
> > -This allows for all relevant parameters to be sliced cleanly during
> > -cross-validation. All other parameters should be passed to
> > -``__init__`` or ``set_params``.
> > +should be restricted to variables that need to be sliced during
> > +cross-validation. Consequently, these variables would be either arrays with shape=[N] or
> > +array-like with shape=[N,D] where N is the number of samples. An example of
> > +this type of variable would be
> > +``sample_weight``. This interface allows all data-dependent
> > +parameters to be sliced cleanly during cross-validation.
> > +All other parameters should be passed to
> > +``__init__`` or set after construction using ``set_params``.
>
> yeah, i was mistaken about GridSearchCV compatibility with set_params, I
> thought it could only construct objects with __init__ and I wasn't
> previously seeing how one could get GridSearchCV to work on params that
> you don't set at __init__.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/7156/files/c0193304c4304f6cba77749faa86b48964466fda..bf626c9de30b0e51a42d8076748eda54cf8a67dd#r73808517>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKNQ_NaIlWLZ2uyr2ybfZ5n3QjYFOqnQks5qdkCJgaJpZM4JeZ4l>
> .
>